authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -576,8 +576,9 @@ public Boolean onGrantedRolesToUser(List<Role> roles, User user)
               try {
                 rangerClient.grantRole(rangerServiceName, grantRevokeRoleRequest);
               } catch (RangerServiceException e) {
-                // Ignore exception, support idempotent operation
-                LOG.warn("Grant role: {} to user: {} failed!", role, user, e);
+                throw new AuthorizationPluginException(
+                    "Fail to grant role %s to user %s, exception: %s",
+                    role.name(), user.name(), e.getMessage());
               }
```

Comment:
Another consideration as to why you don't throw an exception is to support idempotent operations.
Gravitino -> Ranger
Maybe the data was deleted in the last operation of the Ranger, and now it's being operated again. You can't throw an exception.
If you want to throw an exception, you have to cover the idempotent operation, with a test case.

Reply:
You origin test code have covered the logic. You can see RangerHiveIT#testOnGrantedRolesToUser